### PR TITLE
Integerating api

### PR DESCRIPTION
**Pull request by babaralishah**

### CODE REVIEW

**Comment by babaralishah** [comment]
Code organization and structure:
The code is organized into separate folders for different components of the app, such as components, services, and models. This is a good practice as it helps in maintaining and scaling the application.

Naming conventions:
The naming conventions for classes, components, and services are consistent and follow Angular best practices. This makes the code more readable and maintainable.

Code readability:
The code is easy to read and follow. Variable and function names are mostly descriptive, making it easier to understand what the code does.

Security:
The code seems to use best practices to ensure security. It uses bcrypt for password encryption, which is a secure way of storing passwords.

Error handling:
The code has some error handling implemented, but it could be improved. It's important to handle errors properly to prevent unexpected behavior and to make the app more robust.

Code optimization:
The code could be optimized further to improve performance. For example, using lazy loading for modules that are not immediately needed can help speed up the app's initial loading time.

In conclusion, the code follows best practices in terms of naming conventions, organization, and security. However, there is room for improvement in error handling and code optimization to make the app more robust and performant.

**Comment by babaralishah** [comment]
Use arrow functions in component classes to bind methods to 'this':


```
// Before:
export class HomeComponent implements OnInit {
  constructor(private blogService: BlogService) {}

  ngOnInit() {
    this.blogService.getBlogs().subscribe((data) => {
      this.blogs = data;
    });
  }
}

// After:
export class HomeComponent implements OnInit {
  constructor(private blogService: BlogService) {}

  ngOnInit() {
    this.blogService.getBlogs().subscribe(data => {
      this.blogs = data;
    });
  }
}
```


Use interfaces to define data types instead of 'any':

```// Before:
getBlogs(): Observable<any> {
  return this.http.get(this.serverUrl);
}

// After:
interface Blog {
  title: string;
  author: string;
  content: string;
}

getBlogs(): Observable<Blog[]> {
  return this.http.get<Blog[]>(this.serverUrl);
}
```

Use 'const' instead of 'let' for values that don't change:
```// Before:
let serverUrl = 'http://localhost:3000/api/blogs';

// After:
const serverUrl = 'http://localhost:3000/api/blogs';
```

Use 'ng-container' instead of 'div' to wrap multiple elements without adding extra nodes to the DOM:
```// Before:
<div class="row">
  <div class="col-md-6" *ngFor="let blog of blogs">
    <app-blog-card [blog]="blog"></app-blog-card>
  </div>
</div>

// After:
<ng-container class="row">
  <div class="col-md-6" *ngFor="let blog of blogs">
    <app-blog-card [blog]="blog"></app-blog-card>
  </div>
</ng-container>
```

Use 'trackBy' with 'ngFor' to improve performance when rendering lists:

```// Before:
<div class="col-md-6" *ngFor="let blog of blogs">
  <app-blog-card [blog]="blog"></app-blog-card>
</div>

// After:
<div class="col-md-6" *ngFor="let blog of blogs; trackBy: trackById">
  <app-blog-card [blog]="blog"></app-blog-card>
</div>

...

trackById(index: number, blog: Blog) {
  return blog.id;
}
```

Use 'private' or 'public' to specify access modifiers for class members:

```// Before:
blogs: Blog[];

// After:
private blogs: Blog[];
```

Use 'ngOnInit' instead of 'constructor' to initialize class properties:

```// Before:
constructor(private blogService: BlogService) {
  this.blogs = [];
  this.blogService.getBlogs().subscribe((data) => {
    this.blogs = data;
  });
}

// After:
constructor(private blogService: BlogService) {}

ngOnInit() {
  this.blogService.getBlogs().subscribe((data) => {
    this.blogs = data;
  });
}
```
